src/layout.sh

```diff
@@ -156,7 +156,24 @@ start_listener() {
 
         [[ "$source" != "$dest" ]] && __recalculate_layout;
       else
-        __recalculate_layout;
+        desk_file="/tmp/bsp-layout_desktop.txt"
+
+        if [ ! -f "$desk_file" ]; then
+          # create file
+          echo 2 > "$desk_file"
+        fi
+
+        if [[ "$event" == "desktop_focus" ]]; then
+          if [ "$(< "$desk_file")" == "1" ]; then
+            echo 0 > "$desk_file"
+            __recalculate_layout;
+          fi
+        else
+          if [[ "$(< "$desk_file")" == "0" || "$event" != "desktop_focus" ]]; then
```

Comment:
`event` can never be different from `desktop_focus`? :thinking: 
what about `node_add`, `node_remove`, ...? :confused:

Reply:
What I mean is, if event == desktop_focus, it'll go in the `if` above, if it isn't, it'll reach the `else` below. Which means if it is in the `else` block, event is not desktop_focus so that part of the condition is redundant.

Reply:
oh yep i see what you mean!

and the snippet still looks very confusing to me...